src/visualizers/VisualizerGrid.ts

```diff
@@ -0,0 +1,1214 @@
+import {VisualizerExportModule} from '@/visualizers/VisualizerInterface'
+import {VisualizerDefault} from '@/visualizers/VisualizerDefault'
+import {floorSqrt} from '@/shared/math'
+
+// prettier-ignore
+const FIBONACCI_NUMBERS = [
+    0n, 1n, 1n, 2n, 3n, 5n, 8n, 13n, 21n, 34n, 55n, 89n, 144n, 233n,
+    377n, 610n, 987n, 1597n, 2584n, 4181n, 6765n, 10946n, 17711n,
+    28657n, 46368n, 75025n, 121393n, 196418n, 317811n, 514229n, 832040n,
+    1346269n, 2178309n, 3524578n, 5702887n, 9227465n, 14930352n,
+    24157817n, 39088169n, 63245986n, 102334155n, 165580141n, 267914296n,
+    433494437n, 701408733n, 1134903170n, 1836311903n, 2971215073n,
+    4807526976n, 7778742049n, 12586269025n,
+]
+
+// prettier-ignore
+const LUCAS_NUMBERS = [
+    2n, 1n, 3n, 4n, 7n, 11n, 18n, 29n, 47n, 76n, 123n, 199n, 322n, 521n,
+    843n, 1346n, 2207n, 3571n, 5778n, 9349n, 15127n, 24476n, 39603n,
+    64079n, 103682n, 167761n, 271443n, 439204n, 710647n, 1149851n,
+    1860498n, 3010349n, 4870847n, 7881196n, 12752043n, 20633239n,
+    33385282n, 54018521n, 87403803n, 141422324n, 228826127n, 370248451n,
+    599074578n, 969323029n, 1568397607n, 2537720636n, 4106118243n,
+    6643838879n, 10749957122n,
+]
+
+const BLACK = '#000000'
+const WHITE = '#ffffff'
+
+const RED = '#ff5733'
+const ORANGE = '#ff9d33'
+const YELLOW = '#e6ff33'
+const GREEN = '#14cd33'
+const BLUE = '#3388ff'
+const PURPLE = '#8814cd'
+const CYAN = '#00ffff'
+const MAGENTA = '#ff00ff'
+const VERDANT = '#9BBF30'
+const VIOLET = '#7f00ff'
+const MUSTARD = '#ffdb58'
+const GRAY = '#808080'
+
+const HIGHLIGHT = RED
+const SHADE13 = 'rgb(255, 255, 255)'
+const SHADE12 = 'rgb(240, 240, 240)'
+const SHADE11 = 'rgb(220, 220, 220)'
+const SHADE10 = 'rgb(200, 200, 200)'
+const SHADE9 = 'rgb(180, 180, 180)'
+const SHADE8 = 'rgb(160, 160, 160)'
+const SHADE7 = 'rgb(140, 140, 140)'
+const SHADE6 = 'rgb(120, 120, 120)'
+const SHADE5 = 'rgb(100, 100, 100)'
+const SHADE4 = 'rgb(080, 080, 080)'
+const SHADE3 = 'rgb(060, 060, 060)'
+const SHADE2 = 'rgb(040, 040, 040)'
+const SHADE1 = 'rgb(020, 020, 020)'
+const SHADE0 = 'rgb(000, 000, 000)'
+
+enum Preset {
```

Comment:
I'm using enums because they seem to be a simple solution to get rid of holding important data when it can only have a certain set of values, each of which have specific meanings. This is also valuable because checking .equals("primes") may have a type while === Preset.Primes can't have a typo. Does this answer your question? How else would I do this?